### PR TITLE
Update to firebase-admin v5.0.0

### PR DESCRIPTION
**Pull request by laurenzlong**
### Description

Update dependency to firebase-admin v5.0.0, also added shim to transform auth event payload into the new format UserRecord format from firebase-admin.

### CODE REVIEW

File: spec/providers/auth.spec.ts [hunk 1]
```diff
@@ -102,11 +102,67 @@ describe('AuthBuilder', () => {
...
+        },
+        data: {
+          uid: 'abcde12345',
+          email: 'foo@bar.baz',
+          emailVerified: false,
+          displayName: 'My Display Name',
+          photoURL: 'bar.baz/foo.jpg',
+          disabled: false,
+          metadata: {
+              creationTime: '2016-12-15T19:37:37.059Z',
```
Comment (rjhuijsman) [inline spec/providers/auth.spec.ts]: Can you add a small comment to call out the difference between the old and new wire formats here?

**Comment by laurenzlong** [comment]
Made some changes according to our deprecation discussion, @rjhuijsman can you take another look? I also removed the test for "should tolerate missing fields in the payload" since the test events for the other tests is already missing fields.

File: spec/providers/auth.spec.ts [hunk 2]
```diff
@@ -61,90 +61,69 @@ describe('AuthBuilder', () => {
...
+          expect(data.metadata.lastSignInTime).to.equal('2017-01-01T00:00:00.000Z');
+        }),
       ]);
     });
 
-    // This isn't expected to happen in production, but if it does we should
-    // handle it gracefully.
-    it('should tolerate missing fields in the payload', () => {
-      const cloudFunction = auth.user().onCreate((ev: Event<firebase.auth.UserRecord>) => ev.data);
+    it('should still retain createdAt and lastSignedIn', () => {
```
Comment (rjhuijsman) [inline spec/providers/auth.spec.ts]: A comment on what "createdAt" and "lastSignedIn" are and why they should be retained would be useful.

File: spec/providers/auth.spec.ts [hunk 3]
```diff
@@ -61,90 +61,69 @@ describe('AuthBuilder', () => {
...
-          displayName: 'sample user',
-          photoURL: '',
-          disabled: false,
         },
-      } as any;
+      };
 
-      return expect(cloudFunction(event)).to.eventually.deep.equal(event.data);
+      return Promise.all([
+        cloudFunctionCreate(newEvent).then(data => {
```
Comment (rjhuijsman) [inline spec/providers/auth.spec.ts]: We should also expect that the fields with the old name are still present in this case with the new wire format.

Comment (laurenzlong) [inline spec/providers/auth.spec.ts]: done

**Pull request merged**